### PR TITLE
Yao remove text in blue square and add info circle

### PR DESCRIPTION
**Pull request by yaow62023**
# Description
Do not review this PR, all the code is move to a new PR  https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2186 to solve the conflict in the PR. 

(PRIORITY HIGH) Jae: Blue Square Self-scheduler: Remove “(function under development)” and make the FAQ button work. At Profile Page>Below Blue Squares. 
Fixes USER PROFILE COMPONENT Bug# 16 (PRIORITY HIGH)

## Related PRS (if any):
Related PRs https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1542+614, https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1081

## Main changes explained:
1. Remove “(function under development)” text at top of modal
2. Make the FAQ button open a modal that is editable by an Owner
3. created a new modal "components/UserProfile/ScheduleReasonModal/FAQModal.jsx" based on the EditableInfoModal.jsx
4. Add one of those editable “i” icons above at left of the “Schedule Blue Square Reason” button

## How to test:
1. check into current branch
2. do npm install and npm run start:local to run this PR locally3. log as any user and owner user. 
5. go to dashboard→ Profile page→ button "Schedule Blue Square Reason"
6. verify text "(function under development)" is removed.
7. verify function “info circle” (Above the button "Schedule Blue Square Reason") is working, and only Owner account can edit it. 
8. verify function “FAQ Button” is working, the FAQ Button is also only editable by owner account. 

### Before: 
Screenshot 2024-01-13 at 1 11 46 PM

Screenshot 2024-01-13 at 1 11 55 PM

### New function: 
any account:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/189e0db9-3119-4d44-8ca2-4450e263a676

owner account:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/455e2442-e8fd-4a0b-9e02-4044d15426fe

## Note:
Test it with both owner account and other account.

### CODE REVIEW

File: src/components/UserProfile/ScheduleReasonModal/ScheduleReasonModal.jsx [hunk 1]
```diff
@@ -65,6 +66,20 @@ const ScheduleReasonModal = ({
   },[numberOfReasons,infringementsNum])
 // ===============================================================
 
+  //use document.getElementById to get the element of role
```
Comment (swathyjayaseelan) [inline src/components/UserProfile/ScheduleReasonModal/ScheduleReasonModal.jsx:69]: This is causing the issue. Role can be retrieved from the user object.

File: src/components/UserProfile/ScheduleReasonModal/ScheduleReasonModal.jsx [hunk 2]
```diff
@@ -183,9 +197,16 @@ const ScheduleReasonModal = ({
...
-          <Button variant="success" title="Function coming" onClick={handleClose} style={boxStyle}>
-            FAQ
-          </Button>
+        <Button variant="success" title="FAQ" onClick={toggleFaqButtonModal} style={boxStyle}>
+            <FaqButtonModal
+              areaName="blueSquares_FAQ"
+              areaTitle="Blue Squares FAQ"
+              fontSize={24}
+              isPermissionPage={true}
+              role={role}
```
Comment (swathyjayaseelan) [inline src/components/UserProfile/ScheduleReasonModal/ScheduleReasonModal.jsx:206]: role= {user.role}

**Review by swathyjayaseelan — CHANGES_REQUESTED** [review]
It does not work for me. I have added my comments about the lines causing the issue.  

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13806470/80c005b0-4b12-42de-a6e4-37f7e834cfec

File: src/components/UserProfile/ScheduleReasonModal/ScheduleReasonModal.jsx (hunk 1 above)
Comment (yaow62023) [reply src/components/UserProfile/ScheduleReasonModal/ScheduleReasonModal.jsx:69]: Hi, I fixed this bug, please check it again. Thank you so much!

**Review by Pandani07 — DISMISSED** [review]
Works as intended. Could edit only with an owner account


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44672833/1d0b3184-1645-476e-8932-0a53e843681a

**Review by TareqMia — DISMISSED** [review]
Hi @yaow62023, I have reviewed the changes and everything works as expected. I logged in as an Owner and was able to edit the content of the FAQ button and the info circle. I then logged in as a volunteer and was not able to edit the texts. Also, the mentioned "in development" text was not present. Great work on this 👍 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/39747003/0c3940ed-e406-493d-83ed-968a57c95e3e

**Comment by galgalinidhi** [comment]
Hi @yaow62023, I tested the PR using a Volunteer and an Owner account and it works as intended!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44816413/83a968ba-b063-40cd-b0de-775d0dd9f0d6

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I reviewed your PR using both owner and volunteer account, and everything works correctly as intended. Please refer the videos attached. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/f135745c-a314-4530-9ea3-a9846736ec7f


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/7eeb80ed-9e7e-4fd9-9e8e-6f4a31ba3b13

**Review by XiaohanMeng — CHANGES_REQUESTED** [review]
Hi, I tested this PR with the latest development backend, and most features function well. However, I found some discrepancies or issues:
* When using the owner account to edit text, I noticed that clicking 'close' without hitting 'save' still saves the changes, making the two buttons seem functionally identical.
* Upon accessing the volunteer account, instead of seeing the 'Schedule Blue Square Reason' button in blue as shown in your video tutorial, I encountered a yellow button, which is inconsistent.

Please refer to the attached video.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/fe51fab0-77af-416a-bde6-192aeb2d37d7

**Review by Gugor99 — APPROVED** [review]
### Report
Hi Yao,
I checked out your implementation and it works as intended. I logged in as owner and volunteer too and indeed the FAQ and the info button were only editable with the owner account. Well done!
### Screenshots
Cattura1
Cattura2

**Review by yvonnep165 — DISMISSED** [review]
Hi, I have tested the PR with owner and volunteer accounts. It works as intended and only the owner account can edit the info as in the video below:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/103622026/ed249401-45b5-4dbf-87e5-269bf2da0526

**Pull request closed without merging**